src/framework/ui/tab/tabBar.component.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import {
+  TouchableOpacity,
+  View,
+  ViewProps,
+} from 'react-native';
+import {
+  StyledComponentProps,
+  StyleType,
+} from '@kitten/theme';
+import { TabBarIndicator } from './tabBarIndicator.component';
+import { Props as ChildProps } from './tab.component';
+import { toArray } from '../service/common.service';
+
+type ChildElement = React.ReactElement<ChildProps>;
+
+interface TabBarProps {
+  children: ChildElement | ChildElement[];
+  selectedIndex?: number;
+  onSelect?: (index: number) => void;
+}
+
+export type Props = TabBarProps & StyledComponentProps & ViewProps;
+
+interface State {
+  children: ChildElement[];
+}
+
+export class TabBar extends React.Component<Props, State> {
+
+  static defaultProps: Partial<Props> = {
+    selectedIndex: 0,
+  };
+
+  static getDerivedStateFromProps(props: Props): Partial<State> {
+    return {
+      children: toArray(props.children),
+    };
+  }
+
+  public state: State = {
+    children: [],
+  };
+
+  private tabIndicator: React.RefObject<TabBarIndicator> = React.createRef();
+
+  public scrollToIndex(params: { index: number, animated?: boolean }) {
+    this.tabIndicator.current.scrollToIndex(params);
+  }
+
+  public scrollToOffset(params: { offset: number, animated?: boolean }) {
+    this.tabIndicator.current.scrollToOffset(params);
+  }
+
+  private onChildPress = (index: number) => {
+    if (this.props.onSelect) {
+      this.props.onSelect(index);
+    }
+  };
+
+  private renderChild = (element: ChildElement, index: number) => {
+    return (
+      <TouchableOpacity
+        style={{ flex: 1 }}
+        key={index}
+        activeOpacity={1.0}
+        onPress={() => this.onChildPress(index)}>
```

Comment:
can we remove lambda here?

Reply:
There is a need to handle which tab was pressed
Passing `index` directly to `Tab` component and wrapping it to `Touchable` inside it's own implementation could help, but I don't think tab should know it's index